clients/include/flops.hpp

```diff
@@ -720,42 +720,21 @@ constexpr double syrkx_gflop_count<hipblasDoubleComplex>(int n, int k)
 
 /* \brief floating point counts of TRSM */
 template <typename T>
-constexpr double trmm_gflop_count(int m, int n, hipblasSideMode_t side)
+constexpr double trmm_gflop_count(int m, int n, int k)
 {
-    if(HIPBLAS_SIDE_LEFT == side)
-    {
-        return (1.0 * m * n * (m + 1)) / 1e9;
-    }
-    else
-    {
-        return (1.0 * m * n * (n + 1)) / 1e9;
-    }
+    return (1.0 * m * n * (k + 1)) / 1e9;
 }
```

Comment:
I think it is best to get the flop count from the value NOP from the following file http://www.netlib.org/blas/sblas3time.f contains:

            CALL STRMM( SIDE, UPLO, TRANSA, 'N', M, N, ALPHA, A, LDA, B,
     $                  LDB )
            S2 = SECOND( )
            TIME = ( S2 - S1 ) - ( S1 - S0 )
            IF( SIDE.EQ.'L' )THEN
               NOP = M*M*N
            ELSE
               NOP = M*N*N
            END IF

The number of operations for trmm is m*m*n if HIPBLAS_SIDE_LEFT == side, and m*n*n if HIPBLAS_SIDE_LEFT != side

Reply:
This is almost the same as what I have currently. In `testign_trmm.hpp` we have `int K = (side == HIPBLAS_SIDE_LEFT ? M : N);`, so the only difference would be getting rid of the `+ 1`, which I can do.

Reply:
In the flop count from legacy BLAS they keep the higher order term, so m * n * (n+1) = m * n * n + m * n = O(m * n * n) . The calculation of flop (or NOP) is correct in the limit of large size. I find it easiest to keep with the legacy BLAS calculation, if there is any question on the flop count, we can say we got it from the legacy blas code (and it has not been changed since it was first used in 1-July-1988).